src/main/java/com/conveyal/gtfs/loader/JdbcGtfsExporter.java

```diff
@@ -124,26 +127,70 @@ public FeedLoadResult exportTables() {
+                // route status.
+                frequencySelectSql = Table.FREQUENCIES.generateSelectSql(feedIdToExport, Requirement.OPTIONAL) +
+                        Table.FREQUENCIES.generateJoinSql(Table.TRIPS, feedIdToExport) +
+                        Table.TRIPS.generateJoinSql(Table.ROUTES, feedIdToExport, "route_id") +
+                        routesWhereClause;
+            }
+            result.frequencies = export(Table.FREQUENCIES, frequencySelectSql);
+            // Only write "approved" routes using COPY TO with results of select query
+            String routeSelectSql = null;
+            if (fromEditor) {
+                // The filter clause for routes is simple. We're just checking that the route is APPROVED.
+                routeSelectSql = Table.ROUTES.generateSelectSql(feedIdToExport, Requirement.OPTIONAL) +
+                        routesWhereClause;
+            }
+            result.routes = export(Table.ROUTES, routeSelectSql);
             // FIXME: Find some place to store errors encountered on export for patterns and pattern stops.
             // FIXME: Is there a need to export patterns or pattern stops? Should these be iterated over to ensure that
             // frequency-based pattern travel times match stop time arrivals/departures?
 //            export(Table.PATTERNS);
 //            export(Table.PATTERN_STOP);
-            // FIXME: only write shapes for "approved" routes using COPY TO with results of select query
-            result.shapes = export(Table.SHAPES);
+            // Only write shapes for "approved" routes using COPY TO with results of select query
+            String shapeSelectSql = null;
+            if (fromEditor) {
+                // Generate filter SQL for trips if exporting from the editor.
+                // The filter clause for shapes requires two joins to reach the routes table and a where filter on
+                // route status.
+                // FIXME: I'm not sure that shape_id is indexed for the trips table. This could cause slow downs.
+                shapeSelectSql = Table.SHAPES.generateSelectSql(feedIdToExport, Requirement.OPTIONAL) +
+                        Table.SHAPES.generateJoinSql(Table.TRIPS, feedIdToExport) +
+                        Table.TRIPS.generateJoinSql(Table.ROUTES, feedIdToExport, "route_id") +
+                        routesWhereClause;
+            }
+            result.shapes = export(Table.SHAPES, shapeSelectSql);
             result.stops = export(Table.STOPS);
-            // FIXME: only write stop times for "approved" routes using COPY TO with results of select query
-            result.stopTimes = export(Table.STOP_TIMES);
+            // Only write stop times for "approved" routes using COPY TO with results of select query
+            String stopTimesSelectSql = null;
+            if (fromEditor) {
+                // Generate filter SQL for trips if exporting from the editor.
```

Comment:
What does it mean to export "from the editor"? I am guessing this means to export a schema of tables that constitute an editor snapshot, not necessarily from the editor UI. Slightly more specific terminology should be used throughout.